pkgs/common-updater/unstable-updater.nix

```diff
@@ -78,27 +88,51 @@ let
     pushd "$tmpdir"
     commit_date="$(${git}/bin/git show -s --pretty='format:%cs')"
     commit_sha="$(${git}/bin/git show -s --pretty='format:%H')"
-    if [[ -z "$use_stable_version" ]]; then
-        new_version="unstable-$commit_date"
-    else
-        depth=100
-        while (( $depth < 10000 )); do
-            last_tag="$(${git}/bin/git describe --tags --abbrev=0 2> /dev/null || true)"
-            if [[ -n "$last_tag" ]]; then
-                break
+    last_tag=""
+    if [[ -z "$hardcode_zero_version" ]]; then
+        if [[ "$shallow_clone" == "1" ]]; then
+            depth=100
+            while (( $depth < 10000 )); do
+                last_tag="$(${git}/bin/git describe --tags --abbrev=0 2> /dev/null || true)"
+                if [[ -n "$last_tag" ]]; then
+                    tag_matched="$(echo ''${last_tag} | grep -Eo ''${tag_format} || true)"
+                    if [[ -z "$tag_matched" ]]; then
+                        ${lib.getExe git} tag -d ''${last_tag}
+                        continue
+                    fi
+                    break
+                fi
+                ${git}/bin/git fetch --depth="$depth" --tags
+                depth=$(( $depth * 2 ))
+            done
+
+            if [[ -z "$last_tag" ]]; then
+                # To be extra sure, check if full history helps with finding a tag
+                git fetch --tags
+                last_tag="$(${git}/bin/git describe --tags --abbrev=0 2> /dev/null || true)"
             fi
-            ${git}/bin/git fetch --depth="$depth" --tags
-            depth=$(( $depth * 2 ))
-        done
+        else
+            last_tag="$(${git}/bin/git describe --tags --abbrev=0 2> /dev/null || true)"
+        fi
         if [[ -z "$last_tag" ]]; then
-            echo "Cound not found a tag within last 10000 commits" > /dev/stderr
-            exit 1
+            last_tag="0"
         fi
         if [[ -n "$tag_prefix" ]]; then
-          last_tag="''${last_tag#$tag_prefix}"
+            echo "Stripping prefix '$tag_prefix' from tag '$last_tag'"
+            last_tag="''${last_tag#$tag_prefix}"
+        fi
+        if [[ -n "$tag_converter" ]]; then
+            echo "Running '$last_tag' through: $tag_converter"
+            last_tag="$(echo ''${last_tag#$tag_prefix} | ''${tag_converter})"
+        fi
```

Comment:
I find it a bit unintuitive that if both `tag_prefix` and `tag_converter` are provided, that we run them serially.

I think we should have an assert that at most one of these are used.

Reply:
Added an assert so `tagPrefix` and `tagConverter` may not be passed together, and removed the usage of `tag_prefix` in the `tag_converter` case.